File: umpire.rb
strike = 3
ball = 4
gets.to_i.times do
	s = gets.chomp
	if s === "strike"
		strike -= 1
		puts strike > 0 ? "strike!" : "out!"
	elsif s === "ball"
		ball -= 1
		puts ball > 0 ? "ball!" : "fourball!"
	end
end

File: umpire.pl
my $n = <STDIN>;
my $strike = 3;
my $ball = 4;
while ($n--) {
    my $s = <STDIN>;
    chomp($s);
    if ($s eq "strike") {print --$strike > 0 ? "strike!" : "out!";}
    elsif ($s eq "ball") {print --$ball > 0 ? "ball!" : "fourball!";}
    print "\n";
}